src/lib/Tab/LocationView/LocationsTab.php

```diff
@@ -112,12 +143,27 @@ public function getTemplateParameters(array $contextParameters = []): array
         $versionInfo = $content->getVersionInfo();
         $contentInfo = $versionInfo->getContentInfo();
         $locations = [];
+        $pagination = null;
 
         if ($contentInfo->published) {
-            $locationsDataset = $this->datasetFactory->locations();
```

Comment:
I'd double check if `datasetFactory` is being used in the class after you removed this line. If not then we can remove the dependency on it.